apis/nucleus/src/components/listbox/hooks/useExistingModel.jsx

```diff
@@ -15,28 +15,44 @@ export default function useExistingModel({ app, qId, options = {} }) {
   }
 
   useEffect(() => {
-    function storeModel(m) {
-      if (modelStore.get(m.id)) {
-        return;
-      }
-      // Nebula gets the model from this store in some of its
-      // core functions, so it must always be stored here.
-      modelStore.set(m.id, m);
-      m.once('closed', () => {
-        modelStore.clear(m.id);
-      });
-    }
-
+    let isCleaned = false;
+    let cleanupFn = () => {};
     async function fetchObject(modelId) {
       const m = modelStore.get(modelId) || (await app.getObject(modelId));
       return m;
     }
 
-    const p = Promise.resolve(sessionModel || fetchObject(qId));
-    p.then((m) => {
-      storeModel(m);
+    async function awaiter() {
+      await new Promise((resolve) => {
+        setTimeout(() => {
+          resolve();
+        }, 10000);
+      });
+    }
```

Comment:
remove? (I think it was only added to test things)

Reply:
ah yes yes